orchestrator/careplanservice/handle_getservicerequest.go

```diff
@@ -0,0 +1,40 @@
+package careplanservice
+
+import (
+	"context"
+	fhirclient "github.com/SanteonNL/go-fhir-client"
+	"github.com/SanteonNL/orca/orchestrator/lib/auth"
+	"github.com/SanteonNL/orca/orchestrator/lib/coolfhir"
+	"github.com/zorgbijjou/golang-fhir-models/fhir-models/fhir"
+	"net/http"
+)
+
+func (s *Service) handleGetServiceRequest(ctx context.Context, id string, headers *fhirclient.Headers) (*fhir.ServiceRequest, error) {
+	// Verify requester is authenticated
+	_, err := auth.PrincipalFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var serviceRequest fhir.ServiceRequest
+	err = s.fhirClient.Read("ServiceRequest/"+id, &serviceRequest, fhirclient.ResponseHeaders(headers))
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: This query is going to become more expensive as we create more tasks, we should look at setting ServiceRequest.BasedOn or another field to the Task ID
```

Comment:
Shouldn't there only be 1 primary Task referring to this ServiceRequest?

Reply:
Ahh that's a fair point and good to keep in mind. It will be good to cache in the future but yeah not a concern in the meantime